src/sentry/web/frontend/auth_login.py

```diff
@@ -668,6 +668,24 @@ def handle_basic_auth(self, request: Request, **kwargs) -> HttpResponseBase:
                 if not user.is_active:
                     return self.redirect(reverse("sentry-reactivate-account"))
                 if organization:
+                    # Check if the user is a member of the provided organization based on their email
+                    membership = organization_service.check_membership_by_email(
+                        email=user.email, organization_id=organization.id
+                    )
+
+                    invitation_link = getattr(membership, "invitation_link", None)
+
+                    # If the user is a member, the user_id is None, and they are in a "pending invite acceptance" state with a valid invitation link,
+                    # we redirect them to the invitation page to explicitly accept the invite
+                    if (
+                        membership
+                        and membership.user_id is None
+                        and membership.is_pending
+                        and invitation_link
+                    ):
+                        accept_link_position = invitation_link.find("/accept")
+                        return self.redirect(invitation_link[accept_link_position:])
```

Comment:
This may become brittle when that link format changes, you could use urllib here to get the path of the redirect without the domain:
```suggestion
                        accept_path = urllib.parse.urlparse(invitation_link).path
                        return self.redirect(accept_path)
```

Reply:
today I learned! thank you @shellmayr